sound/soc/intel/boards/sdw_rt711_rt1308_rt715.c

```diff
@@ -273,47 +273,42 @@ static int second_spk_init(struct snd_soc_pcm_runtime *rtd)
 	return ret;
 }
 
-SND_SOC_DAILINK_DEF(sdw0_pin2,
-	DAILINK_COMP_ARRAY(COMP_CPU("SDW0 Pin2")));
-SND_SOC_DAILINK_DEF(sdw0_pin3,
-	DAILINK_COMP_ARRAY(COMP_CPU("SDW0 Pin3")));
-SND_SOC_DAILINK_DEF(sdw0_codec,
-	DAILINK_COMP_ARRAY(COMP_CODEC("sdw:0:25d:711:0", "rt711-aif1")));
-
-SND_SOC_DAILINK_DEF(sdw1_pin2,
-	DAILINK_COMP_ARRAY(COMP_CPU("SDW1 Pin2")));
-SND_SOC_DAILINK_DEF(sdw1_codec,
-	DAILINK_COMP_ARRAY(COMP_CODEC("sdw:1:25d:1308:0", "rt1308-aif")));
-
-SND_SOC_DAILINK_DEF(sdw2_pin2,
-	DAILINK_COMP_ARRAY(COMP_CPU("SDW2 Pin2")));
-SND_SOC_DAILINK_DEF(sdw2_codec,
-	DAILINK_COMP_ARRAY(COMP_CODEC("sdw:2:25d:1308:0", "rt1308-aif")));
-
-SND_SOC_DAILINK_DEF(sdw3_pin2,
-	DAILINK_COMP_ARRAY(COMP_CPU("SDW3 Pin2")));
-SND_SOC_DAILINK_DEF(sdw3_codec,
-	DAILINK_COMP_ARRAY(COMP_CODEC("sdw:3:25d:715:0", "rt715-aif2")));
+static int rt1308_i2s_hw_params(struct snd_pcm_substream *substream,
+				struct snd_pcm_hw_params *params)
+{
+	struct snd_soc_pcm_runtime *rtd = substream->private_data;
+	struct snd_soc_card *card = rtd->card;
+	struct snd_soc_dai *codec_dai = rtd->codec_dai;
+	int clk_id, clk_freq, pll_out;
+	int err;
 
-#if IS_ENABLED(CONFIG_SND_SOC_HDAC_HDMI)
-SND_SOC_DAILINK_DEF(idisp1_pin,
-	DAILINK_COMP_ARRAY(COMP_CPU("iDisp1 Pin")));
-SND_SOC_DAILINK_DEF(idisp1_codec,
-	DAILINK_COMP_ARRAY(COMP_CODEC("ehdaudio0D2", "intel-hdmi-hifi1")));
-
-SND_SOC_DAILINK_DEF(idisp2_pin,
-	DAILINK_COMP_ARRAY(COMP_CPU("iDisp2 Pin")));
-SND_SOC_DAILINK_DEF(idisp2_codec,
-	DAILINK_COMP_ARRAY(COMP_CODEC("ehdaudio0D2", "intel-hdmi-hifi2")));
-
-SND_SOC_DAILINK_DEF(idisp3_pin,
-	DAILINK_COMP_ARRAY(COMP_CPU("iDisp3 Pin")));
-SND_SOC_DAILINK_DEF(idisp3_codec,
-	DAILINK_COMP_ARRAY(COMP_CODEC("ehdaudio0D2", "intel-hdmi-hifi3")));
-#endif
+	clk_id = RT1308_PLL_S_MCLK;
+	clk_freq = 38400000;
```

Comment:
The clk_freq is platform dependency.  We may need a quirk if we want to support multi platforms by this driver.

Reply:
Now, it is only for TGL. To make it general we need a quirk.